rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -686,6 +696,65 @@ void getOldFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
 
     }
 
+    @Test
+    void getFederationElection_whenElectionIsInStorage_shouldReturnNewElection() {
+
+        // Arrange
+
+        AddressBasedAuthorizer authorizer = getTestingAddressBasedAuthorizer();
+
+        ABICallElection expectedElection = getSampleElection("function1", authorizer);
+        byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        storageAccessor.saveToRepository(FEDERATION_ELECTION_KEY.getKey(), expectedElectionEncoded);
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        ABICallElection actualElection = federationStorageProvider.getFederationElection(authorizer);
+
+        // Assert
+
+        assertArrayEquals(expectedElectionEncoded, serializeElection(actualElection));
+
+        // Setting a different election in storage to assert that calling the method again returns the cached election
+
+        ABICallElection secondElectionSample = getSampleElection("function2", authorizer);
+        storageAccessor.saveToRepository(FEDERATION_ELECTION_KEY.getKey(), BridgeSerializationUtils.serializeElection(secondElectionSample));
+
+        ABICallElection cachedElection = federationStorageProvider.getFederationElection(authorizer);
+
+        assertArrayEquals(serializeElection(actualElection), serializeElection(cachedElection));
+        assertFalse(Arrays.equals(expectedElectionEncoded, serializeElection(secondElectionSample)));
```

Comment:
this should probably go in another test, wdyt?

Reply:
It would look exactly like this one. For this part of the test, the above part needs to happen first. Meaning, it is dependent on it.

Reply:
i would put it in separated tests anyways, so the test tests what its name describes

Reply:
Updated.